src/storage/src/table/state_table.rs

```diff
@@ -57,9 +57,8 @@ pub struct StateTableBase<S: StateStore, E: Encoding> {
 }
 
 impl<S: StateStore, E: Encoding> StateTableBase<S, E> {
-    /// Note: `dist_key_indices` is ignored, use `new_with[out]_distribution` instead.
-    // TODO: remove this after all state table usages are replaced by `new_with[out]_distribution`.
-    pub fn new(
+    /// Create a state table without distribution, used for tests.
+    pub fn new_for_test(
```

Comment:
Let's remove this and replace all references with `new_without_distribution`?

Reply:
LGTM